lib/cocoapods-core/specification/linter.rb

```diff
@@ -379,11 +384,9 @@ def _validate_deprecated_in_favor_of(d)
         end
       end
 
+      # Performs validations related to the `test_type` attribute.
+      #
       def _validate_test_type(t)
-        unless consumer.spec.test_specification?
```

Comment:
this moved to the `analyzer.rb` and will check all `test_only` attributes that should be part only within test specifications.